src/fstab-generator/fstab-generator.c

```diff
@@ -574,8 +574,33 @@ static int parse_fstab(bool initrd) {
                                 if (streq(canonical_where, where))
                                         canonical_where = mfree(canonical_where);
                                 else
-                                        log_debug("Canonicalized what=%s where=%s to %s",
-                                                  what, where, canonical_where);
+                                        log_debug("Canonicalized where=%s to %s (for what=%s)",
+                                                  where, canonical_where, what);
+                        }
+                }
+
+                if (fstab_test_option(me->mnt_opts, "bind\0" "rbind\0") ||
+                    STR_IN_SET(me->mnt_type, "bind", "rbind")) {
+
+                        _cleanup_free_ char *canonical_what = NULL;
+
+                        path_kill_slashes(what);
```

Comment:
this is should be redundant if we call chase_symlinks() right after anyway, as chase_symlinks() does the same implicitly.